### PR TITLE
Update Language sub-directories of AppInit, Module, and Step

### PR DESCRIPTION
**Pull request by Craig-Elre**
Adding a few tool tips, adding example and corresponding description, expanding explicit descriptions.

### CODE REVIEW

**Comment by jeapostrophe** [comment]
Can you please rebase and use the `@{tooltip(x, y)}` form that I had William add for you? (Instead of writing the HTML directly) --- https://github.com/reach-sh/reach-lang/pull/666

File: docs/src/rsh/module/index.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # {#ref-programs-module} Modules
 
-A Reach @{defn("source file")} is a textual file which specifies a Reach module<a class="ui-tooltip" title="a component of a software program containing one or more routines, it can be a separate file from the main program"><span style="cursor: help;">module</span></a>.
+A Reach @{defn("source file")} is a textual file which specifies a Reach @{tooltip(module, a component of a software program containing one or more routines, it can be a separate file from the main program)}.
```
Comment (jeapostrophe) [inline docs/src/rsh/module/index.md:3]: Did this compile? I thought you needed to add `"`?

Comment (Craig-Elre) [reply docs/src/rsh/module/index.md:3]: Where do I add the quotes? I have put them in a couple different places and none of them compile. I looked at both his example and his code and could not find where to add them. No matter where I put them, I get the error "tooltip is not defined" and I am not sure what that means

Comment (jeapostrophe) [reply docs/src/rsh/module/index.md:3]: @{tooltip("text", "hover")}

Did you rebase?

Comment (Craig-Elre) [reply docs/src/rsh/module/index.md:3]: Bah, you're right...I rebased my branch where I am developing, but did not rebase where I generate the docs.

Comment (Craig-Elre) [reply docs/src/rsh/module/index.md]: Added the "" and verified it built and displayed correctly.

**Review by jeapostrophe — APPROVED** [review]

**Pull request merged**